spackbot/handlers/style.py

```diff
@@ -23,3 +31,99 @@ async def style_comment(event, gh):
+    """
+    response = requests.get(event.data["issue"]["pull_request"]["url"])
+    pr = response.json()
+
+    # Get the sender of the PR - do they have write?
+    sender = event.data["sender"]["login"]
+    repository = event.data["repository"]
+    collaborators_url = repository["collaborators_url"]
+
+    # If they don't have write, we don't allow the command
+    if not await helpers.found(gh.getitem(collaborators_url, {"collaborator": sender})):
+        logger.info(f"Not found: {sender}")
+        return (
+            "Sorry %s, I cannot do that for you. Only users with write can make this request!"
+            % sender
+        )
+
+    # Tell the user the style fix is going to take a minute or two
+    message = "Let me see if I can fix that for you! This might take a moment..."
+    await gh.post(event.data["issue"]["comments_url"], {}, data={"body": message})
+
+    # Get the username of the original committer
+    user = pr["user"]["login"]
+
+    # We need the user id if the user is before July 18. 2017
+    email = helpers.get_user_email(user)
+
+    # We need to use the git url with ssh
+    branch = pr["head"]["ref"]
+    full_name = pr["head"]["repo"]["full_name"]
+    clone_url = "git@github.com:%s.git" % full_name
+
+    # At this point, we can clone the repository and make the change
+    with helpers.temp_dir() as cwd:
+
+        # Clone a fresh spack develop to use for spack style
+        git("clone", "--depth", "1", "https://github.com/spack/spack", "spack-develop")
+
+        spack = sh.Command(f"{cwd}/spack-develop/bin/spack")
+
+        git("clone", "-b", branch, clone_url)
+        os.chdir("spack")
+        git("config", "user.name", user)
+        git("config", "user.email", email)
+
+        # This will authenticate the push with the added ssh credentials
+        git("remote", "set-url", "origin", clone_url)
+
+        # We need to add develop remote for this to work
+        git("remote", "add", "upstream", "https://github.com/spack/spack")
+        git("fetch", "upstream")
+
+        # This won't work if we are already on a develop branch
+        if branch != "develop":
+
+            git("checkout", "--track", "upstream/develop")
+            git("checkout", branch)
+
+        # Save the message for the user
+        res, err = helpers.run_command(spack, ["--color", "never", "style", "--fix"])
+        message = comments.get_style_message(res)
+
+        # Commit (allow for no changes)
+        res, err = helpers.run_command(
+            git,
+            ["commit", "-a", "-m", "[spackbot] updating style on behalf of %s" % user],
+        )
+
+        # Continue differently if the branch is up to date or not
+        if is_up_to_date(res):
+            message += "\nI wasn't able to make any further changes, but please see the message above for remaining issues you can fix locally!"
+            return message
+        message += "\nI've updated the branch with isort fixes."
```

Comment:
Mostly minor but I would write this stuff using the error code exceptions in `sh`:

```suggestion
        try:
            git("commit", "-a", "-m", f"[spackbot] updating style on behalf of {user}")
        except sh.ErrorReturnCode_1 as e:
            # `commit` returns 1 if the branch is up to date
            if not is_up_to_date(e.stdout):
                raise  # something else went wrong
                
            return message + (
                "\nI wasn't able to make any further changes, "
                "but please see the message above for remaining "
                "issues you can fix locally!"
            )
                
        message += "\nI've updated the branch with isort fixes."
```

Reply:
This doesn't seem to work and I'm not liking the "raise" case - I think we always want spackbot to respond with something.

Reply:
Yeah could omit the `raise` -- I modified it above.  I'm fine with ignoring this and using your helper though.